### PR TITLE
Feature/hood maintainer

### PR DESCRIPTION
**Pull request by tly3**

### CODE REVIEW

File: src/main/java/competition/subsystems/hood/HoodSubsystem.java [hunk 1]
```diff
@@ -2,52 +2,85 @@
...
     final DoubleProperty minAngleProp;
+    final DoubleProperty defaultForwardHeadingProp;
+    final DoubleProperty ticksPerDegreeProp;
+    final BooleanProperty calibratedProp;
     final DoubleProperty currentAngleProp;
     private IdealElectricalContract contract;
-    double maxAngle;
-    double minAngle;
     public XCANTalon hoodMotor;
+    double goalAngle;
```
Comment (aschokking) [inline src/main/java/competition/subsystems/hood/HoodSubsystem.java]: you might want these to be DoubleProperty s instead so you can debug what's going on more easily?

File: src/main/java/competition/subsystems/hood/HoodSubsystem.java [hunk 2]
```diff
@@ -2,52 +2,85 @@
...
 import xbot.common.math.MathUtils;
+import xbot.common.properties.BooleanProperty;
 import xbot.common.properties.DoubleProperty;
 import xbot.common.properties.PropertyFactory;
 
 @Singleton
-public class HoodSubsystem extends BaseSubsystem{
+public class HoodSubsystem extends BaseSetpointSubsystem{
 
+    private double calibrationOffset;
```
Comment (aschokking) [inline src/main/java/competition/subsystems/hood/HoodSubsystem.java]: seems like this should be a persistent double property so we can easily tweak it without redeploying?

File: src/main/java/competition/subsystems/hood/HoodSubsystem.java [hunk 3]
```diff
@@ -2,52 +2,85 @@
...
+        setIsCalibrated(true);
+    }
+    
+    public void setIsCalibrated(boolean value){
+        calibratedProp.set(value);
     }
 
-    public void setAngle(double angle){
-        this.angle = angle;
+    public void uncalibrate(){
```
Comment (aschokking) [inline src/main/java/competition/subsystems/hood/HoodSubsystem.java]: what would call this?

File: src/main/java/competition/subsystems/hood/HoodSubsystem.java [hunk 4]
```diff
@@ -2,52 +2,77 @@
...
     }
 
-    public void setAngle(double angle){
-        this.angle = angle;
+    public boolean isCalibrated(){
+        return calibratedProp.get();
+    }
+
+    public void setGoalAngle(double angle){
+        goalAngleProp.set(angle);;
```
Comment (JohnGilb) [inline src/main/java/competition/subsystems/hood/HoodSubsystem.java]: this definitely won't compile - you have two `;`;

File: src/main/java/competition/subsystems/hood/commands/HoodMaintainerCommand.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    final HumanVsMachineDecider decider;
+    final OperatorInterface oi;
+
+    @Inject
+    public HoodMaintainerCommand(HoodSubsystem hood, PropertyFactory pf, PIDFactory pidf, CommonLibFactory clf,
+    OperatorInterface oi){
+        super(hood, pf, clf, 1, 0.33);
+        this.hood = hood;
+        this.oi = oi;
+        decider = clf.createHumanVsMachineDecider(this.getPrefix());
```
Comment (JohnGilb) [inline src/main/java/competition/subsystems/hood/commands/HoodMaintainerCommand.java]: I don't think you need the decider here.

**Review by JohnGilb — CHANGES_REQUESTED** [review]
Let's change the hood to use a 0 to 1 system for "how extended it is" instead of using precise angle for now.

**Review by JohnGilb — APPROVED** [review]

**Pull request merged**